### ISSUE TITLE
Values in `.evaluate()`'s progress bar do not match final output

### ISSUE THREAD

**Issue by markveilletteLL** [keras-team/keras#19934]
For certain metrics (e.g. AUC) comuted in `.evaluate`, the values shown in the progress bar are misleading, and the final state in the progress bar does not reflect the actual metric value.   

Here's a code sample that reproduces the issue,  tested in `keras==3.4.1` using torch backend (but the behavior is the same across backends).

```python
import keras
import numpy as np

# make up some data..
np.random.seed(1234)
X = np.random.rand(1000,32,32,3)
y = np.round(np.mean(X.reshape(1000,-1) ,axis=1)+ 0.1*np.random.randn(1000))

# Define a model
inp = keras.Input(shape=(32,32,3))
x = keras.layers.Conv2D(32,3,activation='relu')(inp)
x = keras.layers.Conv2D(32,3,activation='relu')(inp)
x = keras.layers.Conv2D(32,3,activation='linear')(inp)
x = keras.layers.Flatten()(x)
x = keras.layers.Dense(1)(x)
model = keras.Model(inputs=inp,outputs=x)

# loss and metrics
loss = keras.losses.BinaryCrossentropy( from_logits=True)
metrics = [keras.metrics.AUC(from_logits=True,num_thresholds=2000)]
model.compile(metrics=metrics,loss=loss)

# fit
model.fit(X,y,epochs=1,verbose=0)


# Evaluate using .evaluate
mo = model.evaluate(X,y)
print(f'AUC from .evaluate:  ',mo[1])

# manually evaluate
m = keras.metrics.AUC(from_logits=True,num_thresholds=2000)
yhat = model.predict(X,verbose=0)
m.update_state(y,yhat)
print('Manually computed AUC:',m.result())
```

The output of this code is below.  Note the specific values are RNG dependent, but the point still stands.

```
32/32 ━━━━━━━━━━━━━━━━━━━━ 0s 4ms/step - auc: 0.9291 - loss: 0.5608
AUC from .evaluate:   0.9319638013839722
Manually computed AUC: tensor(0.9320, device='cuda:0')
```

Which shows that the final AUC computed using both `.evaluate` and manually, but the AUC value shown in the progress bar is different.   In some cases the differences are quite large.

**Comment by Grvzard** [comment]
Actually the metric values shown in the progress bar are mean values, i.e. mean AUC of 32 steps.

**Comment by fchollet** [comment]
That metric in the progress bar is a training metric computed, on the training data. There are N different reasons a training metric might turn out different from the metric you see in a call to `evaluate()` (e.g. the state of the model changes at each step, and training behavior might be different from inference behavior).

 If you want your displayed metrics to be equivalent to a call to `evaluate()` done at the end of the epoch, just pass `validation_data=...` in `fit()`. This will call `evaluate()` at the end of the epoch on the data you passed, and it will display the result in the progress bar under the name `"val_auc"`.

**Comment by markveilletteLL** [comment]
> Actually the metric values shown in the progress bar are mean values, i.e. mean AUC of 32 steps.

But the mean of AUC over batches =/= Final AUC.

**Comment by markveilletteLL** [comment]
> That metric in the progress bar is a training metric, computed on the training data. There are N different reasons a training metric might turn out different from the metric you see in a call to `evaluate()` (e.g. the state of the model changes at each step, and training behavior might be different from inference behavior).
> 


AFAIK,  `.evaluate` just takes one dataset so it doesn't distinguish between train/test/val.  I understand the progress bar I see during `.fit` reflects the changing state of the model, however the model weights should be frozen during `evaluate` , and the model should be in inference mode.  So based on what you said I don't see why the metrics in `.evaluate`'s progress bar shouldn't converge to the final metric.  

> Actually the metric values shown in the progress bar are mean values, i.e. mean AUC of 32 steps.

If this is true, and the progress bar shows a *running average* of the metric computed over batches,    then it is understandable the answers are different.  However I still think it's misleading and a better choice would be to show the result of the metric after incorporating each batch. (i.e. the progress bar should show `metric.update_state(...).result()` )
